src/resolution/snapshot.rs

```diff
@@ -903,36 +904,39 @@ pub struct SnapshotFromLockfileParams<'a> {
 /// [`IncompleteSnapshot`] instance that's passed as the first argument for this
 /// function.
 #[allow(clippy::needless_lifetimes)] // clippy bug
-pub async fn snapshot_from_lockfile<'a>(
-  params: SnapshotFromLockfileParams<'a>,
+pub async fn snapshot_from_lockfile<'a, TNpmRegistryApi: NpmRegistryApi>(
```

Comment:
We still need events when resolving one of these from a lockfile, but we can add that in a separate PR.